pluto-message-ingestion/pluto-message-processor.js

```diff
@@ -0,0 +1,95 @@
+class PlutoMessageProcessor {
+  constructor({ hostname, hmacRequest, logger }) {
+    this.hostname = hostname;
+    this.hmacRequest = hmacRequest;
+    this.logger = logger;
+  }
+
+  process(message) {
+    switch (message.type) {
+      case 'project-created': {
+        const project = PlutoMessageProcessor._convertMessageToProject(message);
+
+        return new Promise((resolve, reject) => {
+          this._createProject(project)
+            .then(resp => {
+              this.logger.log('successfully created project', {
+                response: resp
+              });
+              resolve(resp);
+            })
+            .catch(err => {
+              this.logger.error('failed to create project', {
+                status: err.status,
+                response: err.response,
+                project: project
+              });
+              reject(err);
+            });
+        });
+      }
+      case 'project-updated': {
+        const project = PlutoMessageProcessor._convertMessageToProject(message);
+
+        return new Promise((resolve, reject) => {
+          this._updateProject(project)
+            .then(resp => {
+              this.logger.log('successfully updated project', {
+                response: resp
+              });
+              resolve(resp);
+            })
+            .catch(err => {
+              const logDetail = {
+                status: err.status,
+                response: err.response,
+                project: project
+              };
+
+              if (err.status === 404) {
```

Comment:
Thinking about it some more, as we're launching with Sport and News, whose projects span a fortnight at max, I'm tempted to leave this as is as and address it in the future should it become a problem when more desks adopt the Tool, as its the cheapest option.